### PR TITLE
Add unit test for wallet

### PR DESCRIPTION
**Pull request by liuchengxu**
- export and import private key
- update wallet when a block has been attached

### CODE REVIEW

File: blockchain/wallet/wallet_test.go [hunk 1]
```diff
@@ -1,3 +1,219 @@
...
+
+	if len(rawPriv) != 68 {
+		t.Fatal("invalid private key hash length")
+	}
+
+	var xprv [64]byte
+	copy(xprv[:], rawPriv[:64])
+
+	_, err = w.ImportAccountPrivKey(hsm, xprv, xpub.Alias, pwd, 0, acnt1.Alias)
+	if err != pseudohsm.ErrDuplicateKeyAlias {
```
Comment (yahtoo) [inline blockchain/wallet/wallet_test.go]: 缺少导入结果检查。建议检查账户是否创建成功，检查新建账户xpub是否和预期相同。

File: blockchain/wallet/wallet_test.go [hunk 2]
```diff
@@ -1,3 +1,219 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	priv, err := w.ExportAccountPrivKey(hsm, xpub.XPub, pwd)
+
+	rawPriv, err := base58.Decode(*priv)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (yahtoo) [inline blockchain/wallet/wallet_test.go]: 缺少导出结果检查，建议检查导出私钥是否和预期相同。

File: blockchain/wallet/wallet_test.go (hunk 1 above)
Comment (liuchengxu) [reply blockchain/wallet/wallet_test.go]: 👌

File: blockchain/wallet/wallet_test.go (hunk 2 above)
Comment (liuchengxu) [reply blockchain/wallet/wallet_test.go]: 👌

**Pull request merged**